chain/ethevents/events.go

```diff
@@ -127,7 +133,108 @@ func NewEthEvents(contractsAddresses []common.Address, signer *ethereum.SignKeys
 			eventQueue:            make(map[string]*logEvent),
 			eventProcessorRunning: false,
 		},
-	}, nil
+	}
+
+	if scrutinizer != nil {
+		log.Infof("starting ethevents with scrutinizer enabled")
+		ethev.Scrutinizer = scrutinizer
+		ethev.Scrutinizer.AddEventListener(ethev)
+	}
+
+	return ethev, nil
+}
+
+// OnComputeResults is called once a process result is computed by the scrutinizer.
+func (ev *EthereumEvents) OnComputeResults(results *models.ProcessResult) {
+	tctx, cancel := context.WithTimeout(context.Background(), time.Minute*5)
+	defer cancel()
+	// check ethereum
+	// check ethereum process status
+	var fixedSizePID [types.ProcessIDsize]byte
+	copy(fixedSizePID[:], results.ProcessId)
+	ethProcessData, err := ev.VotingHandle.VotingProcess.Get(&ethbind.CallOpts{Context: tctx}, fixedSizePID)
+	if err != nil {
+		log.Errorf("error fetching process %x from Ethereum: %s", results.ProcessId, err)
+		return
+	}
+	if models.ProcessStatus(ethProcessData.Status) == models.ProcessStatus_RESULTS {
+		// check ethereum process results
+		ethResultsData, err := ev.VotingHandle.VotingProcess.GetResults(&ethbind.CallOpts{Context: tctx}, fixedSizePID)
+		if err != nil {
+			log.Errorf("error fetching process %x from Ethereum: %s", results.ProcessId, err)
+			return
+		}
+		if ethResultsData.Height > 0 {
+			log.Errorf("process %x results already added to ethereum", results.ProcessId)
+			return
+		}
+	} else if models.ProcessStatus(ethProcessData.Status) != models.ProcessStatus_ENDED {
+		log.Errorf("invalid process %x status for setting the results", results.ProcessId)
+		return
+	} else {
```

Comment:
if the other two branches always return, you can drop this else and unindent the rest of the code, which will help readability